Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/MainActivity.kt

```diff
@@ -89,6 +115,40 @@ class MainActivity : AppCompatActivity(), HasAndroidInjector {
         vm.isOnboardingDone.observe(this) { isOnboardingDone ->
             startNestedGraphDestination(navController, isOnboardingDone)
         }
+
+        if (savedInstanceState == null) {
+            processExtraParameters()
+        }
+    }
+
+    private fun processExtraParameters() {
+        when (getActionFromIntent(intent)) {
+            MainActivityActions.ADD_DIARY_ENTRY -> {
+                homeFragmentVm.activateContactJournalShortcut()
+            }
+        }
+    }
+
+    fun goToContactJournal() {
+        val navController = supportFragmentManager.findNavController(R.id.nav_host_fragment)
+        findViewById<BottomNavigationView>(R.id.main_bottom_navigation).selectedItemId =
+            R.id.contact_diary_nav_graph
+
+        if (vm.isOnboardingDone.value == true) {
+            val today = LocalDate().toString()
+            navController.navigate(
+                ContactDiaryOverviewFragmentDirections.actionContactDiaryOverviewFragmentToContactDiaryDayFragment(
+                    selectedDay = today
+                )
+            )
+        } else {
+            navController.navigate(
+                ContactDiaryOnboardingFragmentDirections.actionContactDiaryOnboardingFragmentToContactDiaryOnboarding(
+                    showBottomNav = true,
+                    goToCurrentDay = true
+                )
+            )
+        }
```

Comment:
How do you feel about using navigating by Uri in this case with a custom schema such as corona-warn-app:// :
```kotlin
val request = NavDeepLinkRequest.Builder
    .fromUri("android-app://androidx.navigation.app/profile".toUri())
    .build()
findNavController().navigate(request)
```

Reply:
I don't know string URL could be error-prone, what would be the benefit of using the deep link within the app?

Reply:
The benefit is that you can navigate deeply in the graph without having to pass some parameters from a screen into another in order to land on final screen. for example this flag ` goToCurrentDay = true` that you set in order to go the Day screen

Reply:
`goToCurrentDay` parameter means that you should be redirect to `ContactDiaryDayFragment` after onboarding. We always want to show onboarding as a first page. I updated the code with deeplinks, let me know if you consider it better now.